### PR TITLE
fix: withdraw below zero

### LINKED ISSUE #100: Handle withdrawing more money than what the portfolio value is
- [ ] detect when the portfolio value dips below zero and show alert above the graph
- [ ] from the point when the portfolio is below zero, set the graph values should be all set to zero

### LINKED ISSUE #801: Withdrawals are still possible after investment value is zero
I generated a synthetic financial plan with Claude (https://github.com/snaha/kalkul/pull/800) and I found that the app still allows withdrawals after the investment value is zero. 

Image

### LINKED ISSUE #802: Fees can be positive in some cases
I generated a synthetic financial plan with Claude (https://github.com/snaha/kalkul/pull/800) and I found that fees can be positive in some cases. Maybe it is related to #801

Image

### PR DESCRIPTION
**Pull request by agazso**
This PR fixes situations where the investment value could go below zero. This is no longer possible and an error is reported when that would happen. This PR does not implement any UI change, but with the reported errors this is going to be possible once there is a UI design ready for this.

Fixes #801
Fixes #802 
Fixes #100 
Partially addresses #127

### CODE REVIEW

File: src/lib/@snaha/kalkul-maths/investment-calculations.test.ts [hunk 1]
```diff
@@ -671,3 +680,198 @@ describe('#getCurrentPortfolioValue', () => {
 		expect(value).toBeGreaterThanOrEqual(0)
 	})
 })
+
+describe('Error scenarios - investment value below zero protection', () => {
```
Comment (vojtechsimetka) [inline src/lib/@snaha/kalkul-maths/investment-calculations.test.ts]: I am not too excited about this describe system but it seems AI has a tendency to write them: "error scenarios" or "edge cases"...

**Review by vojtechsimetka — APPROVED** [review]

File: src/lib/@snaha/kalkul-maths/investment-calculations.test.ts (hunk 1 above)
Comment (agazso) [reply src/lib/@snaha/kalkul-maths/investment-calculations.test.ts]: I moved them under the `#getInvestmentValues` tests. The best is to tell Claude not to do that and then record that in the CLAUDE.md file.
